msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignInJITTest.kt

```diff
@@ -0,0 +1,250 @@
+import com.microsoft.identity.client.e2e.utils.assertResult
+import com.microsoft.identity.internal.testutils.nativeauth.ConfigType
+import com.microsoft.identity.internal.testutils.nativeauth.api.TemporaryEmailService
+import com.microsoft.identity.internal.testutils.nativeauth.api.models.NativeAuthTestConfig
+import com.microsoft.identity.nativeauth.INativeAuthPublicClientApplication
+import com.microsoft.identity.nativeauth.parameters.NativeAuthChallengeAuthMethodParameters
+import com.microsoft.identity.nativeauth.parameters.NativeAuthGetAccessTokenParameters
+import com.microsoft.identity.nativeauth.parameters.NativeAuthSignInContinuationParameters
+import com.microsoft.identity.nativeauth.parameters.NativeAuthSignInParameters
+import com.microsoft.identity.nativeauth.parameters.NativeAuthSignUpParameters
+import com.microsoft.identity.nativeauth.statemachine.errors.MFASubmitChallengeError
+import com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult
+import com.microsoft.identity.nativeauth.statemachine.results.MFARequiredResult
+import com.microsoft.identity.nativeauth.statemachine.results.RegisterStrongAuthChallengeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignInResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpResendCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpResult
+import com.microsoft.identity.nativeauth.statemachine.states.RegisterStrongAuthVerificationRequiredState
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
+import org.junit.Ignore
+import org.junit.Test
+import org.robolectric.RuntimeEnvironment.application
+import org.robolectric.shadows.ShadowPackageManager.resources
+import org.robolectric.versioning.AndroidVersions
+import java.util.Base64
+
+class SignInJITTest : NativeAuthPublicClientApplicationAbstractTest() {
+
+    private val tempEmailApi = TemporaryEmailService()
+
+    private lateinit var resources: List<NativeAuthTestConfig.Resource>
+
+    lateinit var application: INativeAuthPublicClientApplication
+    lateinit var config: NativeAuthTestConfig.Config
+
+    private val defaultConfigType = ConfigType.SIGN_IN_MFA_SINGLE_AUTH
+    private val defaultChallengeTypes = listOf("password", "oob")
+
+    /**
+     * Full flow: Ensure JIT is triggered on first signIn
+     * - SignUp a new user with username and password.
+     * - Do not SignIn after signUp, but start a new signIn flow.
+     * - Check that JIT flow is triggered.
+     * - Specify a different email as verification contact.
+     * - Complete JIT. Verification email should be sent to the second email.
+     * - Access token is received.
+     *
+     */
+    @Ignore("Retrieving OTP code failure.")
+    @Test
+    fun `test sign in without specify verification contact`()  {
```

Comment:
Can we mention that we are signing up?
```suggestion
    fun `test sign in after sign up without specify verification contact. no second email is excpected`()  {
```

Reply:
Done.